crates/bevy_pbr/src/light.rs

```diff
@@ -52,6 +53,84 @@ impl PointLightUniform {
     }
 }
 
+/// A Directional light.
+///
+/// Directional lights don't exist in reality but they are a good
+/// approximation for light sources VERY far away, like the sun or
+/// the moon.
+#[derive(Debug, Reflect)]
+#[reflect(Component)]
+pub struct DirectionalLight {
+    pub color: Color,
+    pub intensity: f32,
+    direction: Vec3,
+}
+
+impl DirectionalLight {
+    /// Create a new directional light component.
+    ///
+    /// # Panics
+    /// Will panic if `direction` is not normalized.
+    pub fn new(color: Color, intensity: f32, direction: Vec3) -> Self {
+        assert!((direction.length_squared() - 1.0).abs() < 0.0001);
+        DirectionalLight {
+            color,
+            intensity,
+            direction,
+        }
+    }
+
+    /// Set direction of light.
+    ///
+    /// # Panics
+    /// Will panic if `direction` is not normalized.
+    pub fn set_direction(&mut self, direction: Vec3) {
+        assert!((direction.length_squared() - 1.0).abs() < 0.0001);
```

Comment:
Similar here to up above. `assert!` should have a message.

Instead of asserting here we could also create a `NormalizedVec3` struct and have that be the parameter instead of just a `Vec3`.

```rust
pub struct NormalizedVec3(Vec3);

impl NormalizedVec3 {
    pub fn new(vec: Vec3) -> Option<Self>;
}

impl TryFrom<Vec3> for NormalizedVec3 { /**/ }
```